cdk/api.py

```diff
@@ -0,0 +1,187 @@
+                access_log_format=apigw.AccessLogFormat.clf(),
+                data_trace_enabled=True,
+                metrics_enabled=True,
+                logging_level=apigw.MethodLoggingLevel.INFO,
+                stage_name=names.API_STAGE
+            )
+        )
+
+        self.api.add_api_key("ZoomIngesterApiKey")
+
+        self.new_recording_resource = self.api.root.add_resource("new_recording")
+        self.new_recording_method = self.new_recording_resource.add_method(
+            "POST",
+            method_responses=[
+                apigw.MethodResponse(
+                    status_code="200",
+                    response_models={
+                        "application/json": apigw.Model.EMPTY_MODEL
+                    }
+                )
+            ]
+        )
+
+        self.ingest_resource = self.api.root.add_resource("ingest",
+            default_cors_preflight_options=apigw.CorsOptions(
+                allow_origins=apigw.Cors.ALL_ORIGINS,
+                allow_methods=["POST", "OPTIONS"],
+                allow_headers=apigw.Cors.DEFAULT_HEADERS \
+                              + ["Accept-Language","X-Requested-With"]
+            )
+        )
+        on_demand_integration = apigw.LambdaIntegration(on_demand_function)
+        self.ingest_method = self.ingest_resource.add_method(
+            "POST",
+            on_demand_integration,
+            method_responses=[
+                apigw.MethodResponse(
+                    status_code="200",
+                    response_models={
+                        "application/json": apigw.Model.EMPTY_MODEL
+                    }
+                )
+            ]
+        )
+
+        def endpoint_url(resource_name):
+            return (f"https://{self.api.rest_api_id}.execute-api."
+                    f"{core.Stack.of(self).region}.amazonaws.com/"
+                    f"{names.API_STAGE}/{resource_name}")
+
+        on_demand_function.add_environment(
+            "WEBHOOK_ENDPOINT_URL",
+            endpoint_url("new_recording")
+        )
+
+        core.CfnOutput(self, "WebhookEndpoint",
+            export_name=f"{stack_name}-{names.WEBHOOK_ENDPOINT}-url",
+            value=endpoint_url("new_recording")
+        )
+
+        core.CfnOutput(self, "OnDemandEndpoint",
+            export_name=f"{stack_name}-{names.ON_DEMAND_ENDPOINT}-url",
+            value=endpoint_url("ingest")
+        )
+
+        core.CfnOutput(self, "WebhookResourceId",
+            export_name=f"{stack_name}-{names.WEBHOOK_ENDPOINT}-resource-id",
+            value=self.new_recording_resource.resource_id
+        )
+
+        core.CfnOutput(self, "OnDemandResourceId",
+            export_name=f"{stack_name}-{names.ON_DEMAND_ENDPOINT}-resource-id",
+            value=self.ingest_resource.resource_id
+        )
+
+        core.CfnOutput(self, "RestApiId",
+            export_name=f"{stack_name}-{names.REST_API}-id",
+            value=self.api.rest_api_id
+        )
```

Comment:
Putting things like this in cloudformation outputs really makes some of the tasks easier.